elastic/eui
# [EuiInlineEdit] Create `placeholder` Prop

Fixes #6857 

## Summary
Creation of the `displayDefaultValueAsPlaceholder` prop for `EuiInlineEdit`. This is boolean that determines if the string provided by `defaultValue` should display as a placeholder or not.

In `readMode`, the style of the text is updated to a lighter color with italic font. In `editMode`, the form control is passed a placeholder and does not contain the default value. 

If a placeholder is set, it will last until text is successfully saved by inline edit. At that point, the placeholder styling and input element placeholder are both removed.

![Placeholder](https://github.com/elastic/eui/assets/40739624/381d6407-e35d-4e08-8f52-153fb53761a0)


## QA

Head over to `EuiInlineEdit` in [staging](https://eui.elastic.co/pr_6872/#/forms/inline-edit#setting-a-placeholder) and ensure you can do the following:
- [ ] Distinguish the difference between placeholder stying and default styling in `readMode`
- [ ] See the placeholder text on the edit mode form control after clicking on the read mode button
- [ ] Head to edit mode as if you're going to change the text. Cancel your edit and select the read mode button again. The placeholder should still be visible.
- [ ] Head to edit mode, make a tex change, and save. Select the read mode button again and you should find your text populated in the form control.

Remove or strikethrough items that do not apply to your PR.

### General checklist
